Makefile

```diff
@@ -1,198 +1,336 @@
 #!/usr/bin/make -f
 
-PREFIX=/opt/yandex/ch-tools
-INSTALL_DIR=$(DESTDIR)$(PREFIX)
+ifndef VERBOSE
+.SILENT:
+endif
+
+SHELL := bash
+.ONESHELL:
+.SHELLFLAGS := -eu -o pipefail -c
+.DELETE_ON_ERROR:
+MAKEFLAGS += --warn-undefined-variables
+MAKEFLAGS += --no-builtin-rules
+
+PYTHON ?= python3
+
+# The latest version supporting Python 3.6
+POETRY_VERSION ?= 1.1.15
+POETRY_HOME ?= /opt/poetry
+POETRY := $(POETRY_HOME)/bin/poetry
+
+PREFIX ?= /opt/yandex/ch-tools
+BUILD_PYTHON_OUTPUT_DIR ?= dist
+BUILD_DEB_OUTPUT_DIR ?= out
+SRC_DIR ?= ch_tools
+TESTS_DIR ?= tests
+
+# It is used by DEB building tools to install a program to temporary
+# directory before packaging
+DESTDIR ?=
+
+INSTALL_DIR = $(DESTDIR)$(PREFIX)
+BIN_DIR = $(INSTALL_DIR)/bin
+SYMLINK_BIN_DIR = $(DESTDIR)/usr/bin
+
+WHL_FILE = ch_tools-*.whl
+VENV_DIR = .venv
+VERSION_FILE = version.txt
+INSTALL_DEPS_STAMP = .install-deps
+
+# Pass arguments for testing tools
+BEHAVE_ARGS ?= -D skip_setup
+PYTEST_ARGS ?=
+
 
 .PHONY: install
 install: install-python-package install-symlinks install-bash-completions configure-logs ;
 
+
 .PHONY: uninstall
 uninstall: uninstall-python-package uninstall-symlinks uninstall-bash-completions uninstall-logrotate ;
 
+
+.PHONY: install-deps
+install-deps: $(INSTALL_DEPS_STAMP) ;
+
+
+# Update dependencies in poetry.lock to their latest versions according to your pyproject.toml
+.PHONY: update-deps
+update-deps:
+	$(POETRY) update
+
+
+$(INSTALL_DEPS_STAMP): ensure-poetry venv pyproject.toml 
```

Comment:
Shouldn't `poetry.lock` be in dependency list?

Reply:
Yes, it is more robust. I also did some refinements.